lib/data_food_consortium/connector/importer.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require_relative "skos_parser"
+
+module DataFoodConsortium
+  module Connector
+    class Importer
+      TYPES = [
+        DataFoodConsortium::Connector::CatalogItem,
+        DataFoodConsortium::Connector::Enterprise,
+        DataFoodConsortium::Connector::Offer,
+        DataFoodConsortium::Connector::Person,
+        DataFoodConsortium::Connector::QuantitativeValue,
+        DataFoodConsortium::Connector::SuppliedProduct,
+      ].freeze
+
+      def self.type_map
+        @type_map ||= TYPES.each_with_object({}) do |clazz, result|
+          # Methods with variable arguments have a negative arity of -n-1
+          # where n is the number of required arguments.
+          number_of_required_args = -1 * (clazz.instance_method(:initialize).arity + 1)
+          args = Array.new(number_of_required_args)
+          type_uri = clazz.new(*args).semanticType
+          result[type_uri] = clazz
```

Comment:
I trying to get my head around this, you are instantiating classes from `TYPES` so you can get the associated `semanticType`, correct ?

Reply:
Yes, that's the only way the library was exposing the type. I'm sending code changes upstream to make this easier.